mod/dagmodifier.go

```diff
@@ -200,6 +200,20 @@ func (dm *DagModifier) Sync() error {
 	// Number of bytes we're going to write
 	buflen := dm.wrBuf.Len()
 
+	node, ok := dm.curNode.(*mdag.ProtoNode)
+	if ok {
+		fsn, err := ft.FSNodeFromBytes(node.Data())
+		if err != nil {
+			return err
+		}
+		fs := fsn.FileSize()
```

Comment:
You can use the `fileSize` function (above this one) to avoid repeating code and that also supports the raw-node case.

Reply:
Thx for reviewing @schomatis , can you explain more about how raw node can get `filesize`?

> You can use the `fileSize` function (above this one) to avoid repeating code 

where i use repeating code?

Reply:
Thx for reviewing @schomatis , can you explain more about how raw node can get `filesize`?

> You can use the `fileSize` function (above this one) to avoid repeating code 

where will casue repeating code?

Reply:
Thx for reviewing @schomatis , can you explain more about how raw node can get `filesize`?

> You can use the `fileSize` function (above this one) to avoid repeating code 

where will casue repeating code?

Reply:
Thx for reviewing @schomatis , can you explain more about how raw node can get `filesize`?

> You can use the `fileSize` function (above this one) to avoid repeating code 

where will casue repeating code?

Reply:
Thx for reviewing @schomatis , can you explain more about how raw node can get `filesize`?

> You can use the `fileSize` function (above this one) to avoid repeating code 

where will casue repeating code?

Reply:
Thx for reviewing @schomatis , can you explain more about how raw node can get `filesize`?

> You can use the `fileSize` function (above this one) to avoid repeating code 

where will casue repeating code?

Reply:
Thx for reviewing @schomatis , can you explain more about how raw node can get `filesize`?

> You can use the `fileSize` function (above this one) to avoid repeating code 

where will casue repeating code?

Reply:
I was referring to this [`fileSize`](https://github.com/ipfs/go-unixfs/blob/master/mod/dagmodifier.go#L172) function, you can just feed it the `dm.curNode` as the [`DagModifier.Size`](https://github.com/ipfs/go-unixfs/blob/master/mod/dagmodifier.go#L162) does.